docs/architecture/adr-047-handling-evidence-from-light-client.md

```diff
@@ -19,161 +20,200 @@ the light client detector receives two conflicting headers
+This queries the header of the last height for each of the witnesses in order to compare hashes. In the event that these two hashes are 
+different we have verified conflicting headers and therefore the detector proceeds to run 
 
-### Validating headers
+```golang
+FindLightFork(trace []*SignedHeader, witness Provider) (commonHeader, h1, h2 *SignedHeader, error)
+```
 
-Check headers are valid (`ValidateBasic`), are in order of ascending height, and do not exceed the `MaxTraceSize`.
+to verify the witness's header and then locate the Light Bifurcation Point
 
-### Finding Block Bifurcation
+![](../imgs/light-client-detector.png)
 
-The node pulls the block ID's for the respective heights of the trace headers from its own block store.
+This is done by:
 
-First it checks to see that the first header hash matches its first `BlockID` else it can discard it. 
+1. Checking that the trusted header is the same. Currently, they should not theoretically be different
+because witnesses cannot be added and removed after the client is initialized. But we do this any way
+as a sanity check. If this fails we have to drop the witness.
 
-If the last header hash matches the nodes last `BlockID` then it can also discard it on the assumption that a fork can not remerge and hence this is just a trace of valid headers. 
+2. Querying and verifying the witness's headers using bisection at the same heights of all the
+intermediary headers of the primary (In the above example this is A, B, C, D, F, H). If bisection fails or the witness stops responding then 
+we can call the witness faulty and drop it. 
 
-The node then continues to loop in descending order checking that the headers hash doesn't match it's own blockID for that height. Once it reaches the height that the block ID matches the hash it then sends the common header, the trusted header and the diverged header (common header is needed for lunatic evidence) to determine if the divergence is a real offense to the tendermint protocol or if it is just fabricated.
+3. We eventually reach a verified header by the witness which is not the same as the intermediary header (In the above example this is E).
+This is the point of bifurcation (This could also be the last header).
 
-### Figuring out if malicious behaviour
+Finding this point of bifurcation could have come about in two ways:
 
-The node first examines the case of a lunatic attack:
+1. The light client could directly verify between the common header and the diverged header
 
-* The validator set of the common header must have at least 1/3 validator power that signed in the divergedHeaders commit
+2. The light client needs to verify further headers in between the two intermediary heights (Header D and E in the example below).
 
-* One of the deterministically derived hashes (`ValidatorsHash`, `NextValidatorsHash`, `ConsensusHash`,
-`AppHash`, or `LastResultsHash`) of the header must not match:
+![](../imgs/bifurcation-point.png)
```

Comment:
This figure is super useful. It could potentially be more clear if instead of letters, height numbers are used. Then it would be clear where at what heights we have diverging headers.

Reply:
Sure I can quickly modify it